src/shared.ts

```diff
@@ -0,0 +1,84 @@
+import {Stripe, StripeConstructor} from '@stripe/stripe-js';
+
+// `_VERSION` will be rewritten by `@rollup/plugin-replace` as a string literal
+// containing the package.json version
+declare const _VERSION: string;
+
+const V3_URL = 'https://js.stripe.com/v3';
+
+const injectScript = (): HTMLScriptElement => {
+  const script = document.createElement('script');
+  script.src = V3_URL;
+
+  const headOrBody = document.head || document.body;
+
+  if (!headOrBody) {
+    throw new Error(
+      'Expected document.body not to be null. Stripe.js requires a <body> element.'
+    );
+  }
+
+  headOrBody.appendChild(script);
+
+  return script;
+};
+
+const registerWrapper = (stripe: any): void => {
+  if (!stripe || !stripe._registerWrapper) {
+    return;
+  }
+
+  stripe._registerWrapper({name: 'stripe-js', version: _VERSION});
+};
+
+export const loadScript = (): Promise<StripeConstructor | null> => {
+  // Execute our own script injection after a tick to give users time to
+  // do their own script injection.
+  const stripePromise: Promise<StripeConstructor | null> = Promise.resolve().then(
```

Comment:
Suggestion: make `stripePromise` a top level `let stripePromise;`, and make `loadScript` a little less stateful? You could also `export let stripePromise;` for helping fully reset between tests?

Reply:
That seems nicer, updated!